src/components v2/Atoms/Components/Figure/Figure.vue

```diff
@@ -0,0 +1,145 @@
+<template>
+  <div class="photon-figure" :class="[size]">
+    <span class="figure-container">
+      <div
+        class="figure"
+        :class="[$attrs.class, (!loading && 'animate-fade-in-slow') || '']"
+      >
+        {{ loading && isCurrency ? calculation : display }}
+      </div>
+    </span>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue, { PropType } from 'vue';
+import { addCommaSeparators } from '@/utils';
+import { FiguresData } from './types';
+import { formProps } from '@/components v2/Atoms/Forms/globalProps';
+
+const { size } = formProps;
+
+export const props = {
+  value: {
+    type: Number as PropType<number>,
+    default: 0,
+  },
+  loading: {
+    type: Boolean as PropType<boolean>,
+    default: false,
+  },
+  isCurrency: {
+    type: Boolean as PropType<boolean>,
+    default: true,
+  },
+  size,
+};
+
+export default Vue.extend({
+  name: 'P2Figure',
+  props,
+
+  data(): FiguresData {
+    return {
+      calculation: 0.0,
+      calcInterval: 0,
+      min: 0,
+      max: 0,
+    };
+  },
+  computed: {
+    display() {
+      return (this as any).isCurrency
+        ? `$${addCommaSeparators(this.value)}`
+        : (this as any).value;
+    },
```

Comment:
this might be good to get @DanDroryNano opinion if we want to extract this computed property to the OB-web parent?

Reply:
Leave it here for now. we're going to get rid of this component in the long run.